Framework/Algorithms/src/ReflectometryReductionOne2.cpp

```diff
@@ -239,30 +581,81 @@ ReflectometryReductionOne2::makeDirectBeamWS(MatrixWorkspace_sptr inputWS) {
   return directBeamWS;
 }
 
+/**
+* Normalize the workspace by the direct beam (optional)
+*
+* @param detectorWS : workspace in wavelength which is to be normalized
+* @return : corrected workspace
+*/
+MatrixWorkspace_sptr ReflectometryReductionOne2::directBeamCorrection(
+    MatrixWorkspace_sptr detectorWS) {
+
+  MatrixWorkspace_sptr normalized = detectorWS;
+  Property *directBeamProperty = getProperty("RegionOfDirectBeam");
+  if (!directBeamProperty->isDefault()) {
+    const auto directBeam = makeDirectBeamWS(m_runWS);
+    normalized = divide(detectorWS, directBeam);
+  }
+
+  return normalized;
+}
+
+/**
+* Perform either transmission or algorithmic correction according to the
+* settings.
+* @param detectorWS : workspace in wavelength which is to be normalized
+* @param detectorWSReduced:: whether the input detector workspace has been
+* reduced
+* @return : corrected workspace
+*/
+MatrixWorkspace_sptr ReflectometryReductionOne2::transOrAlgCorrection(
+    MatrixWorkspace_sptr detectorWS, const bool detectorWSReduced) {
+
+  MatrixWorkspace_sptr normalized;
+  MatrixWorkspace_sptr transRun = getProperty("FirstTransmissionRun");
+  if (transRun) {
+    normalized = transmissionCorrection(detectorWS, detectorWSReduced);
+  } else if (getPropertyValue("CorrectionAlgorithm") != "None") {
```

Comment:
These string values are also used in ReflectometryWorkflowBase2.cpp, so this is not so straightforward.